src/kokkos/KokkosCore/HistoContainer.h

```diff
@@ -0,0 +1,339 @@
+#ifndef HeterogeneousCore_CUDAUtilities_interface_HistoContainer_h
+#define HeterogeneousCore_CUDAUtilities_interface_HistoContainer_h
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <type_traits>
+
+#include "KokkosCore/AtomicPairCounter.h"
+
+using team_policy = Kokkos::TeamPolicy<KokkosExecSpace>;
+using member_type = Kokkos::TeamPolicy<KokkosExecSpace>::member_type;
+
+namespace cms {
+  namespace kokkos {
+    template <typename Histo, typename T>
+    KOKKOS_FUNCTION void countFromVector(Kokkos::View<Histo*,KokkosExecSpace> h,
+                                         const uint32_t nh,
+                                         Kokkos::View<T const*,KokkosExecSpace> v,
+                                         Kokkos::View<uint32_t const*,KokkosExecSpace> offsets,
+                                         const member_type& teamMember) {
+      uint32_t first = teamMember.league_rank() * teamMember.team_size() + teamMember.team_rank();
+      uint32_t total_threads = teamMember.league_size() * teamMember.team_size();
+      for (uint32_t i = first, nt = offsets(nh); i < nt; i += total_threads) {
+        uint32_t index = 0;
+        for(uint32_t j = 0;j<=nh;++j){
+          if(offsets(j) > i){
+             index = j;
+             break;
+          }
+        }
```

Comment:
I don't think `cuda_std::upper_bound()` would work directly since we have a Kokkos::View object instead of a real pointer to an array. Unless there is a way to get the actual pointer to the array in memory, perhaps there is an I'm not familiar with it yet.

Reply:
`offsets.data()` gives a pointer to the underlying array (even though its use is discouraged IIRC).

Reply:
I'll just write an `upper_bound` that operates on the view if that is OK. That will clarify the code and keep the kokkos methodology intact.

Reply:
An `upper_bound()` with `Kokkos::View` would be fine as well, thanks.